cli/src/config.rs

```diff
@@ -283,17 +284,46 @@ impl UnresolvedConfigEnv {
 
         if let Some(path) = legacy_platform_config_path {
             if path.exists() {
+                Self::warn_for_deprecated_path(
+                    ui,
+                    path.as_path(),
+                    "~/Library/Application Support",
+                    "~/.config",
+                );
                 paths.push(path);
             }
         }
         if let Some(path) = legacy_platform_config_dir {
             if path.exists() {
+                Self::warn_for_deprecated_path(
+                    ui,
+                    path.as_path(),
+                    "~/Library/Application Support",
+                    "~/.config",
+                );
                 paths.push(path);
             }
         }
 
         paths
     }
+
+    fn warn_for_deprecated_path(ui: &Ui, path: &Path, old: &str, new: &str) {
+        let _ = writeln!(
+            ui.warning_default(),
+            "Deprecated configuration file `{}`:",
```

Comment:
nit: maybe `.` instead of `:`?

Each warning line is printed separately with the `Warning: ` header. You can also use `indoc::writedoc!` to format multi-line text cleanly.